heroku/cli
# feat(domains): update custom domains functionality

## Description
[Work Item](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001kE9miYAC/view)
[Work Item](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001kENI3YAO/view)
[Work Item](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001kEK8sYAG/view)

This PR updates and introduces a few new changes for custom domains and the Heroku CLI in general.
- Adds a pagination utility function that can be used to paginate any list request
- Adds `keyValueParser` utility function to parse any `@oclif/core` flags that may contain spaces such as `./bin/run domains -a testing-deploy --filter 'Domain Name=ztestdomain7'`
- Warns users that the application they're attempting to display custom domains for contains over 100 domains
- Prompts users after aforementioned warning with a question on whether they would like to display the full list of custom domains

## Testing
**NOTE**: In order to test these updates, you'll need an app with over 1,000 domains. If you don't have one, please ping me to add you to the test app, `testing-deploy`.

1. Pull down current branch
2. `yarn` it up
3. Check that the new warning and prompt functionality work via 
`./bin/run domains -a testing-deploy`
4. Confirm the new warning and prompt appear like so:
 ```
 ›   Warning: This app has over 100 domains. Your terminal may not be configured to display the total
 ›   amount of domains. You can export all domains into a CSV file with: heroku domains -a example-app
 ›   --csv > example-file.csv
 Display all 1003 domains? (Y/N)
```
5. Confirm that pressing enter auto selects `no` as the default option
6. Run the command again and type `y`. Confirm that all domains are shown.
7. Create a `csv` file by running the example command via
   `./bin/run domains -a testing-deploy --csv > test-domains-file.csv`
8. Confirm all domains are outputted to the `csv` file
9. Apply a filter to the domains command via
    `./bin/run domains -a testing-deploy --filter 'Domain Name=ztestdomain7'`
10. Confirm the prompt asks to display all 111 domains
11. Apply a filter to the domains command that will return less than 100 domains via
    `./bin/run domains -a testing-deploy --filter 'Domain Name=sara'`
12. Confirm that no warning or prompt is displayed and that there is only 1 custom domain
